GenderPayGap.WebUI/Views/ManageOrganisations/UsersRegisteredToReportForOrganisation.cshtml

```diff
@@ -1,35 +1,130 @@
+            {
+                new TableCellViewModel
+                {
+                    Html = @<text>
+                               <p class="govuk-!-margin-top-1 govuk-!-margin-bottom-1">
+                                   @userRegisteredToReport.Fullname
+                                   @if (userRegisteredToReport.UserId == Model.User.UserId)
+                                   {
+                                       <span>(You)</span>
+                                   }
+                                </p>
+                               <p class="govuk-!-margin-top-1 govuk-!-margin-bottom-1">
+                                   <a class="govuk-link" href="@(Url.Action("RemoveOrganisation", "Organisation", new {orgId = encryptedOrganisationId, userId = encryptedUserId}))">
+                                       Remove user <span class="govuk-visually-hidden">@userRegisteredToReport.Fullname</span>
+                                   </a>
+                               </p>
+                            </text>
+                },
+                new TableCellViewModel
+                {
+                    Text = " "
+                }
+            },
+        };
+        mobileRows.Add(newRow);
+    }
+    
+    var desktopRows = new List<TableRowViewModel>();
+    @foreach (User userRegisteredToReport in Model.GetFullyRegisteredUsersForOrganisationWithCurrentUserFirst())
+    {
+        var encryptedUserId = Encryption.EncryptQuerystring(userRegisteredToReport.UserId.ToString());
+        var newRow = new TableRowViewModel
+        {
+            Row = new List<TableCellViewModel>
+            {
+                new TableCellViewModel
+                {
+                    Html = @<text>
+                               @userRegisteredToReport.Fullname
+                               @if (userRegisteredToReport.UserId == Model.User.UserId)
+                               {
+                                   <span>(You)</span>
+                               }
+                               <br/>
+                            </text>
+                },
+                new TableCellViewModel
+                {
+                    Html = @<text>
+                               <a class="govuk-link" href="@(Url.Action("RemoveOrganisation", "Organisation", new {orgId = encryptedOrganisationId, userId = encryptedUserId}))">
+                                  Remove user <span class="govuk-visually-hidden">@userRegisteredToReport.Fullname</span>
+                              </a>
+                            </text>
+                }
+            },
+        };
+        desktopRows.Add(newRow);
+    }
+    
+}
+
+<h2 class="govuk-heading-l">Registered users</h2>
+    @{
+        var usersRegisteredToReportForOrganisation = Model.Organisation.UserOrganisations.Where(uo => uo.PINConfirmedDate.HasValue).Select(uo => uo.User).Distinct().ToList();
+    }
+
+    @if (usersRegisteredToReportForOrganisation.Count == 0)

```

Comment:
Is this the correct logic? Surely it's always at least one, the person viewing this page?

Reply:
I'm pretty sure it doesn't function as intended however I also do not know how it was intended to work, this code was legacy and moved from the view into this partial.
I will create a ticket for someone to take a look at it after confirming how it's use is intended.